doc/whats_new.rst

```diff
@@ -157,342 +149,442 @@ Enhancements
+   - Fix a bug in cases where ``numpy.cumsum`` may be numerically unstable,
+     raising an exception if instability is identified. :issue:`7376` and
+     :issue:`7331` by `Joel Nothman`_ and :user:`yangarbiter`.
 
-   - Fix bug where stratified CV splitters did not work with
-     :class:`linear_model.LassoCV`. :issue:`8973` by
-     :user:`Paulo Haddad <paulochf>`.
+   - Fix a bug where :meth:`base.BaseEstimator.__getstate__`
+     obstructed pickling customizations of child-classes, when used in a
+     multiple inheritance context.
+     :issue:`8316` by :user:`Holger Peters <HolgerPeters>`.
 
-   - Fixed a bug in :class:`linear_model.RandomizedLasso`,
-     :class:`linear_model.Lars`, :class:`linear_model.LassoLars`,
-     :class:`linear_model.LarsCV` and :class:`linear_model.LassoLarsCV`,
-     where the parameter ``precompute`` were not used consistently across
-     classes, and some values proposed in the docstring could raise errors.
-     :issue:`5359` by `Tom Dupre la Tour`_.
+   - Update Sphinx-Gallery from 0.1.4 to 0.1.7 for resolving links in
+     documentation build with Sphinx>1.5 :issue:`8010`, :issue:`7986` by
+     :user:`Oscar Najera <Titan-C>`
 
-   - Fixed a bug where :func:`model_selection.validation_curve`
-     reused the same estimator for each parameter value.
-     :issue:`7365` by :user:`Aleksandr Sandrovskii <Sundrique>`.
 
-   - :class:`multiclass.OneVsOneClassifier`'s ``partial_fit`` now ensures all
-     classes are provided up-front. :issue:`6250` by
-     :user:`Asish Panda <kaichogami>`.
+API changes summary
+-------------------
 
-   - Fixed an integer overflow bug in :func:`metrics.confusion_matrix` and
-     hence :func:`metrics.cohen_kappa_score`. :issue:`8354`, :issue:`7929`
-     by `Joel Nothman`_ and :user:`Jon Crall <Erotemic>`.
+Trees and ensembles
 
-   - Made default kernel parameters kernel-dependent in :class:`kernel_approximation.Nystroem`
-     :issue:`5229` by :user:`mth4saurabh` and `Andreas Müller`_.
+   - Gradient boosting base models are no longer estimators. By `Andreas Müller`_.
 
-   - Fixed passing of ``gamma`` parameter to the ``chi2`` kernel in
-     :func:`metrics.pairwise_kernels` :issue:`5211` by :user:`nrhine1`,
-     :user:`mth4saurabh` and `Andreas Müller`_.
+   - All tree based estimators now accept a ``min_impurity_decrease``
+     parameter in lieu of the ``min_impurity_split``, which is now deprecated.
+     The ``min_impurity_decrease`` helps stop splitting the nodes in which
+     the weighted impurity decrease from splitting is no longer alteast
+     ``min_impurity_decrease``.  :issue:`8449` by `Raghav RV`_.
 
-  -  Fixed a bug in :class:`gaussian_process.GaussianProcessRegressor`
-     when the standard deviation and covariance predicted without fit
-     would fail with a unmeaningful error by default.
-     :issue:`6573` by :user:`Quazi Marufur Rahman <qmaruf>` and
-     `Manoj Kumar`_.
+Linear, kernelized and related models
```

Comment:
Maybe we should have one section name "Neighborhood models" and another "Semi-supervised learning models" instead?